tests/unit/io/Cereal_unittest.cc

```diff
@@ -2,38 +2,55 @@
 #include <shogun/base/Parameter.h>
 #include <gtest/gtest.h>
 
+#include <fstream>
+#include <cstdio>
+
 #include <shogun/lib/SGVector.h>
 #include <cereal/archives/json.hpp>
-#include <fstream>
 
 using namespace shogun;
 
 TEST(Cereal, Json_SGVector_load_equals_saved)
 {
 	const index_t size = 5;
-	SGVector<float64_t> a(size), b;
+	SGVector<float64_t> a(size);
+	SGVector<float64_t> b;
 	a.range_fill(1.0);
 
-	{
-		std::ofstream os("cerealtest_sgvector.cereal");
-		cereal::JSONOutputArchive archive(os);
-		archive(a);
-	}
+	char name[256] = "cerealtest_sgvector.cereal";
+	char* file = tmpnam (name);
 
+	try
 	{
-		std::ifstream is("cerealtest_sgvector.cereal");
-		cereal::JSONInputArchive archive(is);
-		archive(b);
+		{
+			std::ofstream os(file);
+			cereal::JSONOutputArchive archive(os);
+			archive(a);
+		}
+
+		{
+			std::ifstream is(file);
+			cereal::JSONInputArchive archive(is);
+			archive(b);
+		}
+
+		EXPECT_EQ(b.ref_count(), a.ref_count());
+		for (index_t i = 0; i < size; i++)
+			EXPECT_NEAR(a[i], b[i], 1E-15);
+
+		std::remove(file);
 	}
-	
-	for (index_t i = 0; i < size; i++)
-		EXPECT_NEAR(a[i], b[i], 1E-15);
+	catch (const std::ios_base::failure& e)
```

Comment:
when an exception happens, the `std::remove` will not be called, so you will need to call it in the catch section as well.
no need to narrow down to io exceptions only, catch all of them by:
```catch (std::exception& e)```